### PR TITLE
Add back info about VFS storage driver

### LINKED ISSUE #3904: Storage driver documentation lacks vfs
The documentation on storage drivers doesn't list `vfs`. This led a user to ask me if we removed support for `vfs` (and to be fair, he was trying to start the engine with `--storage-driver vfs` and it was still starting with `overlay2` for other reasons, so I can't blame him!)

I think we should re-add explanations about `vfs`!

File: [engine/userguide/storagedriver/selectadriver.md](https://docs.docker.com/engine/userguide/storagedriver/selectadriver/), CC @mstanleyjones

💖,

### PR DESCRIPTION
**Pull request by mstanleyjones**
Fixes #3904 

cc/ @jpetazzo @cpuguy83

### CODE REVIEW

File: engine/userguide/storagedriver/vfs-driver.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
+---
+description: Learn how to optimize your use of VFS driver.
+keywords: 'container, storage, driver, vfs'
+title: Use the VFS storage driver
+---
+
+The VFS storage driver is a simple union filesystem where each layer is a
```
Comment (cpuguy83) [inline engine/userguide/storagedriver/vfs-driver.md]: Is not a union filesystem.

File: engine/userguide/storagedriver/vfs-driver.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    ```
+
+Docker is now using the `vfs` storage driver. Docker has automatically
+created the `/var/lib/docker/vfs/` directory, which contains all the layers
+used by running containers.
+
+
+## How the `vfs` storage driver works
+
+VFS is a *union filesystem*, which means that it layers multiple directories on
```
Comment (cpuguy83) [inline engine/userguide/storagedriver/vfs-driver.md]: Not a union filesystem

File: engine/userguide/storagedriver/selectadriver.md [hunk 3]
```diff
@@ -84,18 +84,22 @@ For Docker CE, only some configurations are tested, and your operating system's
...
-| Docker CE on Ubuntu | `aufs`, `devicemapper`, `overlay2` (Ubuntu 14.04.4 or later, 16.04 or later), `overlay`, `zfs` |
-| Docker CE on Debian | `aufs`, `devicemapper`, `overlay2` (Debian Stretch), `overlay`                                 |
-| Docker CE on CentOS | `devicemapper`                                                                                 |
-| Docker CE on Fedora | `devicemapper`, `overlay2` (Fedora 26 or later, experimental), `overlay` (experimental)        |
+| Linux distribution  | Recommended storage drivers                                                                           |
+|:--------------------|:------------------------------------------------------------------------------------------------------|
+| Docker CE on Ubuntu | `aufs`, `devicemapper`, `overlay2` (Ubuntu 14.04.4 or later, 16.04 or later), `overlay`, `zfs`, `vfs` |
+| Docker CE on Debian | `aufs`, `devicemapper`, `overlay2` (Debian Stretch), `overlay`, `vfs`                                 |
+| Docker CE on CentOS | `devicemapper`, `vfs`                                                                                 |
```
Comment (mstanleyjones) [inline engine/userguide/storagedriver/selectadriver.md]: Look at the issue this was fixing to see the discussion there, please.

File: engine/userguide/storagedriver/selectadriver.md [hunk 4]
```diff
@@ -84,18 +84,22 @@ For Docker CE, only some configurations are tested, and your operating system's
...
+| Docker CE on Fedora | `devicemapper`, `overlay2` (Fedora 26 or later, experimental), `overlay` (experimental), `vfs`        |
 
 When in doubt, the best all-around configuration is to use a modern Linux
 distribution with a kernel that supports the `overlay2` storage driver, and to
 use Docker volumes for write-heavy workloads instead of relying on writing data
 to the container's writable layer.
 
+The `vfs` storage driver is usually not the best choice. Before using the `vfs`
```
Comment (mstanleyjones) [inline engine/userguide/storagedriver/selectadriver.md]: Again look at the linked issue.

**Pull request merged**